app/src/main/java/com/cyberbot/checkers/game/Grid.java

```diff
@@ -89,7 +87,15 @@ public GridEntry getEntryByCoords(int x, int y) throws IndexOutOfBoundsException
     }
 
     public boolean moveAllowed(GridEntry src, GridEntry dst) {
-        return src == dst || calculateAllowedMoves(src, false).contains(dst);
+        if(src == dst) return true;
+
+        ArrayList<CaptureChain> allowedCaptures = getAllowedCaptures(src, true);
+        if(!allowedCaptures.isEmpty()) {
+            return allowedCaptures.stream().anyMatch(
+                    (capture) -> capture.getLocationAfterCapture() == dst
+            );
+        }
+        else return getAllowedMoves(src, true).contains(dst);
     }
 
     public boolean attemptMove(GridEntry src, GridEntry dst) {
```

Comment:
The attempt move should no longer return whether or not the move was allowed, but whether or not the player can execute a next move.

Reply:
A good way to do it would probably be to internally split `moveAllowed` into `ordinaryMoveAllowed` and `captureAllowed`. The method would then expect a call to `moveAllowed` before `atemptMove` and is an attempt is made to move into an invalid location throw an `InvalidMoveException`.